kernels/portable/cpu/op_grid_sampler_2d.cpp

```diff
@@ -0,0 +1,257 @@
+//
+//  Copyright (c) 2025 Apple Inc. All rights reserved.
+//  Provided subject to the LICENSE file in the top level directory.
+//
+
+#include <executorch/kernels/portable/cpu/scalar_utils.h>
+#include <executorch/kernels/portable/cpu/util/elementwise_util.h>
+#include <executorch/runtime/kernel/kernel_includes.h>
+
+#include <algorithm>
+#include <cmath>
+#include <cstdint>
+#include <utility>
+
+#include <c10/util/irange.h>
+
+namespace torch {
+namespace executor {
+namespace native {
+
+using Tensor = exec_aten::Tensor;
+using ScalarType = executorch::aten::ScalarType;
+using SizesType = executorch::aten::SizesType;
+
+// Transform normalized coordinates to pixel space
+inline float unnormalize_coord(float coord, int64_t size, bool align_corners) {
+  if (align_corners) {
+    // -1 and 1 correspond to the centers of the first and last pixels
+    return ((coord + 1.0f) / 2.0f) * (size - 1);
+  } else {
+    // -1 and 1 correspond to the boundary of the image
+    return ((coord + 1.0f) * size - 1.0f) / 2.0f;
+  }
+}
+
+// Compute source index and interpolation weight
+inline std::pair<int64_t, float>
+compute_source_index_and_weight(float coord, int64_t size, bool align_corners) {
+  float real_coord = unnormalize_coord(coord, size, align_corners);
+  int64_t index = std::floor(real_coord);
+  float weight = real_coord - index;
+  return {index, weight};
+}
+
+// Apply reflective padding to handle out-of-bounds coordinates
+inline int64_t reflect_coord(int64_t coord, int64_t size) {
+  if (size <= 1)
+    return 0;
+
+  int64_t double_size = 2 * size - 2;
+  if (double_size <= 0)
+    return 0;
+
+  // Handle negative coordinates
+  int64_t abs_coord = std::abs(coord);
+  abs_coord = abs_coord % double_size;
```

Comment:
super nit: I see a couple places in this PR where you could be terser by performing more than one operation per statement. for example, here would be better as `int64_t abs_coord = std::abs(coord) % double_size;`